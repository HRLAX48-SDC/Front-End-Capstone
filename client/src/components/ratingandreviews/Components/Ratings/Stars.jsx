import React, { useState, useEffect} from 'react';
import Card from 'react-bootstrap/Card'


function Stars (props) {

    return (
      <Card>
        <Card.Title>
            *average*  *star display*
        </Card.Title>
        <Card.Body>
          <Card.Text>
           100% of reviews recommend this product
          </Card.Text>
        </Card.Body>
      </Card>
    )

}
export default Stars;