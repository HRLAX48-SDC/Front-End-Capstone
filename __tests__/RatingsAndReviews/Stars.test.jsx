import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Stars from '../../client/src/components/ratingandreviews/Components/Ratings/Stars.jsx'

describe('<Stars />', () =>{
    const ratings = {1: "23", 2: "8", 3: "50", 4: "36", 5: "47"};


    it('Should correctly render the star rating', () => {
        const stars = renderer
        .create(<Stars ratings={ratings}/>)
        .toJSON();
        expect(stars).toMatchSnapshot()
    })
})