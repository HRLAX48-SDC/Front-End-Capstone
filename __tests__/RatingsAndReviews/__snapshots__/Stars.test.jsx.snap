// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stars /> Should correctly render the star rating 1`] = `
Array [
  <div
    className="hstack gap-3"
  >
    <div
      className="average-review-header"
    >
      0.0
    </div>
    <span
      style={
        Object {
          "direction": "ltr",
          "display": "inline-block",
          "touchAction": "none",
        }
      }
    >
      <span
        aria-hidden="true"
        className="react-simple-star-rating"
        style={
          Object {
            "cursor": "",
            "display": "inline-block",
            "overflow": "hidden",
            "position": "relative",
            "userSelect": "none",
            "verticalAlign": "middle",
            "whiteSpace": "nowrap",
          }
        }
      >
        <span
          className="empty-icons"
          style={
            Object {
              "color": "#cccccc",
              "display": "inline-block",
            }
          }
        >
          <svg
            className="star-svg"
            fill="currentColor"
            height={35}
            style={Object {}}
            viewBox="0 0 24 24"
            width={35}
          >
            <path
              d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
              fill="currentColor"
              stroke="none"
              strokeMiterlimit="10"
              strokeWidth={0}
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={35}
            style={Object {}}
            viewBox="0 0 24 24"
            width={35}
          >
            <path
              d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
              fill="currentColor"
              stroke="none"
              strokeMiterlimit="10"
              strokeWidth={0}
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={35}
            style={Object {}}
            viewBox="0 0 24 24"
            width={35}
          >
            <path
              d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
              fill="currentColor"
              stroke="none"
              strokeMiterlimit="10"
              strokeWidth={0}
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={35}
            style={Object {}}
            viewBox="0 0 24 24"
            width={35}
          >
            <path
              d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
              fill="currentColor"
              stroke="none"
              strokeMiterlimit="10"
              strokeWidth={0}
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={35}
            style={Object {}}
            viewBox="0 0 24 24"
            width={35}
          >
            <path
              d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
              fill="currentColor"
              stroke="none"
              strokeMiterlimit="10"
              strokeWidth={0}
            />
          </svg>
        </span>
        <span
          className="filled-icons"
          style={
            Object {
              "color": "#ffbc0b",
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "transition": "",
              "whiteSpace": "nowrap",
              "width": "0%",
            }
          }
          title="0 out of 5"
        >
          <svg
            className="star-svg"
            fill="currentColor"
            height={35}
            style={Object {}}
            viewBox="0 0 24 24"
            width={35}
          >
            <path
              d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
              fill="currentColor"
              stroke="none"
              strokeMiterlimit="10"
              strokeWidth={0}
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={35}
            style={Object {}}
            viewBox="0 0 24 24"
            width={35}
          >
            <path
              d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
              fill="currentColor"
              stroke="none"
              strokeMiterlimit="10"
              strokeWidth={0}
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={35}
            style={Object {}}
            viewBox="0 0 24 24"
            width={35}
          >
            <path
              d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
              fill="currentColor"
              stroke="none"
              strokeMiterlimit="10"
              strokeWidth={0}
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={35}
            style={Object {}}
            viewBox="0 0 24 24"
            width={35}
          >
            <path
              d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
              fill="currentColor"
              stroke="none"
              strokeMiterlimit="10"
              strokeWidth={0}
            />
          </svg>
          <svg
            className="star-svg"
            fill="currentColor"
            height={35}
            style={Object {}}
            viewBox="0 0 24 24"
            width={35}
          >
            <path
              d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
              fill="currentColor"
              stroke="none"
              strokeMiterlimit="10"
              strokeWidth={0}
            />
          </svg>
        </span>
      </span>
    </span>
  </div>,
  <h5
    className="total-ratings-header"
  >
     ratings
  </h5>,
]
`;
